pnpbridge/src/adapters/src/impinj_reader_r700/impinj_reader_r700.c

```diff
@@ -0,0 +1,666 @@
+    if ((iothubClientResult = PnpBridgeClient_SendEventAsync(clientHandle, EventMessageHandle,
+            EventConfirmationCallback, UserContextCallback)) != IOTHUB_CLIENT_OK)
+    {
+        LogError("IoTHub client call to _SendEventAsync failed with error code %d", iothubClientResult);
+        goto exit;
+    }
+    else
+    {
+        LogInfo("IoTHub client call to _SendEventAsync succeeded");
+    }
+
+exit:
+    return iothubClientResult;
+}
+
+IOTHUB_CLIENT_RESULT 
+ImpinjReader_SendTelemetryMessagesAsync(
+    PNPBRIDGE_COMPONENT_HANDLE PnpComponentHandle
+    )
+{
+    IOTHUB_CLIENT_RESULT result = IOTHUB_CLIENT_OK;
+    IOTHUB_MESSAGE_HANDLE messageHandle = NULL;
+    PIMPINJ_READER device = PnpComponentHandleGetContext(PnpComponentHandle);
+
+    // read curl stream 
+    
+    int uSecInit = clock();
+    int uSecTimer = 0;
+    int uSecTarget = 5000;
+
+    while (uSecTimer < uSecTarget) {  // pull messages out of buffer for target time, then return to calling function
+
+        CURL_Stream_Read_Data read_data = curlStreamReadBufferChunk(device->curl_stream_session);
+
+        if (read_data.dataChunk == NULL) {  // if no data in buffer, stop reading and return to calling function
+            // LogInfo("No data returned from stream buffer.");
+            IoTHubMessage_Destroy(messageHandle);
+            return IOTHUB_CLIENT_OK;
+        }
+
+        #define MESSAGE_SPLIT_DELIMITER "\n\r"
+
+        char * oneMessage = strtok(read_data.dataChunk, MESSAGE_SPLIT_DELIMITER);  // split data chunk by \n\r in case multiple reader events are contained in the same chunk
+            
+        int count = 0;
+
+        while (oneMessage != NULL) {  // send each event individually
+
+            count++;
+
+            LogInfo("TELEMETRY Message %d: %s", count, oneMessage);
+  
+            char * currentMessage = ImpinjReader_CreateJsonResponse("streamReadEvent", oneMessage); // TODO: Parameterize this property name
+
+            if ((messageHandle = PnP_CreateTelemetryMessageHandle(device->SensorState->componentName, currentMessage)) == NULL)
+            {
+                LogError("Impinj Reader Adapter:: PnP_CreateTelemetryMessageHandle failed.");
+            }
+            else if ((result = ImpinjReader_RouteSendEventAsync(PnpComponentHandle, messageHandle,
+                    ImpinjReader_TelemetryCallback, device)) != IOTHUB_CLIENT_OK)
+            {
+                LogError("Impinj Reader Adapter:: SampleEnvironmentalSensor_RouteSendEventAsync failed, error=%d", result);
+            }
+
+            oneMessage = strtok(NULL, MESSAGE_SPLIT_DELIMITER);  // continue splitting until all messages are sent individually
+        }  
+
+        uSecTimer = clock() - uSecInit;
+        // LogInfo("Stream Read Timer: %d, Stream Read Target %d", (int)uSecTimer, (int)uSecTarget);
+        // LogInfo(" TELEMETRY: %s", read_data.dataChunk);
+    }
+
+    IoTHubMessage_Destroy(messageHandle);
```

Comment:
>IoTHubMessage_Destroy(messageHandle); [](start = 4, length = 37)

Destroy the message handle for each telemetry message handle created inside the while loop